File: DrakborgenDotNet/Card.cs
using SFML.Graphics;
using SFML.System;

namespace Drakborgen
{
    internal class Card : Transformable, Drawable
    {
        private Sprite _sprite;

        internal Card(string imagePath)
        {
            _sprite = new Sprite(new Texture(imagePath));
        }

        private void CenterAround(Vector2f position)
        {
            FloatRect bounds = GetGlobalBounds();
            Position = new Vector2f(position.X - (float)(bounds.Width) / 2.0f, position.Y - (float)(bounds.Height) / 2.0f);
        }

        internal FloatRect GetGlobalBounds()
        {
            return Transform.TransformRect(_sprite.GetLocalBounds());
        }

        void Drawable.Draw(RenderTarget target, RenderStates states)
        {
            states.Transform *= Transform;
            target.Draw(_sprite, states);
        }
    }
}

File: src/Card.cpp
#include "Card.h"

Card::Card(const std::string& imagePath)
	: sprite(imagePath)
	, imagePath(imagePath)
{
}

/*Card::Card(Card&& other) noexcept
{
	*this = std::move(other);
}

Card& Card::operator=(Card&& other) noexcept
{
	sprite = std::move(other.sprite);
	return *this;
}*/

void Card::centerAround(sf::Vector2f position)
{
	sf::FloatRect bounds = getGlobalBounds();
	setPosition({ position.x - float(bounds.width) / 2.0f, position.y - float(bounds.height) / 2.0f });
}

sf::FloatRect Card::getGlobalBounds()
{
	return getTransform().transformRect(sprite.get().getLocalBounds());
}

void Card::draw(sf::RenderTarget& target, sf::RenderStates states) const
{
	states.transform *= getTransform();
	target.draw(sprite.get(), states);
}
